File: tests/workflow/producer/rust/src/producer/events/structa.rs
use super::{identification, producer::Control, protocol, Context};
use crate::test::samples;
use clibri::server;
use uuid::Uuid;

type BroadcastStructB = (Vec<Uuid>, protocol::StructB);
type BroadcastStructC = (Vec<Uuid>, protocol::StructC);

#[allow(unused_variables)]
pub async fn emit<E: server::Error, C: server::Control<E> + Send + Clone>(
    event: protocol::StructA,
    filter: &identification::Filter,
    context: &mut Context,
    control: &Control<E, C>,
) -> Result<(BroadcastStructB, BroadcastStructC), String> {
    Ok((
        (filter.all(), samples::struct_b::get()),
        (filter.all(), samples::struct_c::get()),
    ))
}

File: experimental/workflow_mt/producer/typescript/src/producer/events/structa.ts
import { Filter, Producer, Context, Protocol } from "../implementation/events";
import { Output } from "../implementation/events/structa";

export function emit(
    event: Protocol.StructA,
    filter: Filter,
    context: Context,
    producer: Producer
): Promise<Output> {
    return Promise.reject(
        new Error(`Handler for event "StructA" isn't implemented`)
    );
}